pkg/kube/client/upbound/upbound.go

```diff
@@ -0,0 +1,92 @@
+package upbound
+
+import (
+	"context"
+	"net/http"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/upbound/up-sdk-go"
+	"github.com/upbound/up-sdk-go/service/auth"
+	"golang.org/x/oauth2"
+	"k8s.io/client-go/rest"
+
+	"github.com/crossplane-contrib/provider-kubernetes/pkg/kube/client/token"
+)
+
+const (
+	// TODO: We may want to make is configurable.
+	authHost = "auth.upbound.io"
+)
+
+// WrapRESTConfig configures the supplied REST config to use OAuth2 access
+// tokens fetched using the supplied Upbound session/robot token.
+func WrapRESTConfig(_ context.Context, rc *rest.Config, token string, store *token.ReuseSourceStore) error { // nolint:gocyclo // mostly error handling
+	ex := rc.ExecProvider
+	if ex == nil {
+		return errors.New("an identity configuration was specified but the provided kubeconfig does not have execProvider section")
+	}
+	if ex.APIVersion != "client.authentication.k8s.io/v1" {
+		return errors.New("execProvider APIVersion is not client.authentication.k8s.io/v1")
+	}
+	if ex.Command != "up" || len(ex.Args) < 2 || ex.Args[0] != "organization" || ex.Args[1] != "token" {
+		return errors.New("execProvider command is not up organization token")
+	}
+
+	// Read the org name, it could either be the 3rd argument and provided with `ORGANIZATION` env var.
+	org := ""
+	if len(ex.Args) > 2 {
```

Comment:
this does not error if we have more than 3, but I guess that's ok?
if we want, we can return error if len > 3

(or more nicely):
```
if !(len(ex.Args) == 2 || len(ex.Args) == 3) then error
```

Reply:
Already relying on the fact that org is either the 3rd argument or provider with an env var. Erroring out otherwise, i.e. `organization name not provided in execProvider args or ORGANIZATION env var`. 

I don't see much value in being more strict here TBH, and would like to assume the command structure doesn't change (which would be breaking anyways).